File: articles/ego-consciousness-and-psychic-differentiation/fortran/ego_differentiation_scores.f90
! Ego, Consciousness, and Psychic Differentiation
! Fortran example: simple synthetic ego-differentiation score

program ego_differentiation_scores_demo
  implicit none

  real :: coherence
  real :: strain

  coherence = 0.72 * 0.80 - 0.48 * 0.54 + 0.60 * 0.72 - 0.22 * 0.08
  strain = 0.40 * 0.08 + 0.46 * 0.54 + 0.42 * 0.54 + 0.34 * 0.20 - 0.44 * 0.72

  print *, "Synthetic ego coherence:", coherence
  print *, "Synthetic psychic strain:", strain
end program ego_differentiation_scores_demo

File: articles/ego-consciousness-and-psychic-differentiation/go/ego_differentiation_scores.go
package main

import "fmt"

func egoCoherence(
	differentiation,
	unconsciousPressure,
	reflectiveFlexibility,
	oneSidedness float64,
) float64 {
	return 0.72*differentiation -
		0.48*unconsciousPressure +
		0.60*reflectiveFlexibility -
		0.22*oneSidedness
}

func psychicStrain(
	oneSidedness,
	unconsciousPressure,
	shadowActivation,
	egoRigidity,
	reflectiveFlexibility float64,
) float64 {
	return 0.40*oneSidedness +
		0.46*unconsciousPressure +
		0.42*shadowActivation +
		0.34*egoRigidity -
		0.44*reflectiveFlexibility
}

func main() {
	coherence := egoCoherence(0.80, 0.54, 0.72, 0.08)
	strain := psychicStrain(0.08, 0.54, 0.54, 0.20, 0.72)

	fmt.Printf("Synthetic ego coherence: %.3f\n", coherence)
	fmt.Printf("Synthetic psychic strain: %.3f\n", strain)
}
